applications/IgaApplication/custom_modelers/iga_modeler.cpp

```diff
@@ -0,0 +1,279 @@
+        }
+
+        KRATOS_INFO_IF("CreateQuadraturePointGeometries", mEchoLevel > 0)
+            << "Creating " << name << "s of type: " << type
+            << " for " << rGeometryList.size() << " geometries"
+            << " in " << rModelPart.Name() << "-SubModelPart." << std::endl;
+
+        for (SizeType i = 0; i < rGeometryList.size(); ++i)
+        {
+            GeometriesArrayType geometries;
+            rGeometryList[i].CreateQuadraturePointGeometries(
+                geometries, shape_function_derivatives_order);
+
+            KRATOS_INFO_IF("CreateQuadraturePointGeometries", mEchoLevel > 1)
+                << geometries.size() << " quadrature point geometries have been created." << std::endl;
+
+            if (type == "element" || type == "Element") {
+                SizeType id = 1;
+                if (rModelPart.GetRootModelPart().Elements().size() > 0)
+                    id = rModelPart.GetRootModelPart().Elements().back().Id() + 1;
+
+                this->CreateElements(
+                    geometries.ptr_begin(), geometries.ptr_end(),
+                    rModelPart, name, id, PropertiesPointerType());
+            }
+            else if (type == "condition" || type == "Condition") {
+                SizeType id = 1;
+                if (rModelPart.GetRootModelPart().Conditions().size() > 0)
+                    id = rModelPart.GetRootModelPart().Conditions().back().Id() + 1;
+
+                this->CreateConditions(
+                    geometries.ptr_begin(), geometries.ptr_end(),
+                    rModelPart, name, id, PropertiesPointerType());
+            }
+            else {
+                KRATOS_ERROR << "\"type\" does not exist: " << type
+                    << ". Possible types are \"element\" and \"condition\"." << std::endl;
+            }
+        }
+    }
+
+    ///@}
+    ///@name CAD functionalities
+    ///@{
+
+    void IgaModeler::GetCadGeometryList(
+        GeometriesArrayType& rGeometryList,
+        ModelPart& rModelPart,
+        const Parameters rParameters) const
+    {
+        if (rParameters.Has("brep_id")) {
+            rGeometryList.push_back(rModelPart.pGetGeometry(rParameters["brep_id"].GetInt()));
+        }
+        if (rParameters.Has("brep_ids")) {
+            for (SizeType i = 0; i < rParameters["brep_ids"].size(); ++i) {
+                rGeometryList.push_back(rModelPart.pGetGeometry(rParameters["brep_ids"][i].GetInt()));
+            }
+        }
+        if (rParameters.Has("brep_name")) {
+            rGeometryList.push_back(rModelPart.pGetGeometry(rParameters["brep_name"].GetString()));
+        }
+        if (rParameters.Has("brep_names")) {
+            for (SizeType i = 0; i < rParameters["brep_names"].size(); ++i) {
+                rGeometryList.push_back(rModelPart.pGetGeometry(rParameters["brep_names"][i].GetString()));
+            }
+        }
```

Comment:
I think the nameS with the trailing S can be unified. So that we just use "brep_id" or "brep_ids" independetly if there is just one or more. This can be done in the future.

Reply:
yeah, I agree. Thanks for pointing it out.
But, I am also a fan of allowing as many options for typos as possible.